rest-service/manager_rest/resource_manager.py

```diff
@@ -2184,6 +2188,26 @@ def _workflow_queued(self, execution):
 
         send_event(event, 'hook')
 
+    def _create_deployment_initial_dependencies(self,
+                                                deployment_plan,
+                                                source_deployment):
+        new_dependencies = deployment_plan.setdefault(
+            INTER_DEPLOYMENT_FUNCTIONS, {})
+
+        for func_id, target_deployment in new_dependencies.items():
+            target_deployment_instance = None
+            if target_deployment:
+                target_deployment_instance = self.sm.get(models.Deployment,
```

Comment:
I don't see how this would be an improvement of the current implementation. Wouldn't we still have to call `self.sm.get` on each `target_deployment_id`?